sway/input/seatop_move_tiling.c

```diff
@@ -127,27 +201,53 @@ static void handle_motion_postthreshold(struct sway_seat *seat) {
 		return;
 	}
 
+	// Check if the cursor is over a tilebar only if the destination
+	// container is not a descendant of the source container.
+	if (!surface && !container_has_ancestor(con, e->con) &&
+			split_titlebar(node, e->con, cursor->cursor,
+			&e->drop_box, &e->after_target)) {
+		e->target_edge = WLR_EDGE_NONE;
+		// Don't allow dropping over the source container's titlebar
+		// to give users a chance to cancel a drag operation.
+		if (con == e->con) {
+			e->target_node = NULL;
+			return;
```

Comment:
Nit: let's drop this return and just put the fall-through path in an `else`, so that the `e->target_node` assignments mirror their indentation level.